src/commands/captureDiagnostics.ts

```diff
@@ -0,0 +1,281 @@
+import { Writable } from "stream";
+import { WorkspaceContext } from "../WorkspaceContext";
+import { Version } from "../utilities/version";
+import { execFileStreamOutput } from "../utilities/utilities";
+import configuration from "../configuration";
+
+export async function captureDiagnostics(
+    ctx: WorkspaceContext,
+    allowMinimalCapture: boolean = true
+) {
+    try {
+        const captureMode = await captureDiagnosticsMode(ctx, allowMinimalCapture);
+
+        // dialog was cancelled
+        if (!captureMode) {
+            return;
+        }
+
+        const diagnosticsDir = path.join(
+            tmpdir(),
+            `vscode-diagnostics-${formatDateString(new Date())}`
+        );
+
+        await fs.mkdir(diagnosticsDir);
+        await writeLogFile(diagnosticsDir, "extension-logs.txt", extensionLogs(ctx));
+        await writeLogFile(diagnosticsDir, "settings.txt", settingsLogs(ctx));
+        await writeLogFile(diagnosticsDir, "diagnostics.txt", diagnosticLogs());
+
+        if (captureMode === "Full") {
+            // The `sourcekit-lsp diagnose` command is only available in 6.0 and higher.
+            if (ctx.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0))) {
+                await sourcekitDiagnose(ctx, diagnosticsDir);
+            } else {
+                await writeLogFile(diagnosticsDir, "sourcekit-lsp.txt", sourceKitLogs(ctx));
+            }
+        }
+
+        ctx.outputChannel.log(`Saved diagnostics to ${diagnosticsDir}`);
+        await showCapturedDiagnosticsResults(diagnosticsDir);
+    } catch (error) {
+        vscode.window.showErrorMessage(`Unable to capture diagnostic logs: ${error}`);
+    }
+}
+
+export async function promptForDiagnostics(ctx: WorkspaceContext) {
+    const ok = "OK";
+    const cancel = "Cancel";
+    const result = await vscode.window.showInformationMessage(
+        "SourceKit-LSP has been restored. Would you like to capture a diagnostic bundle to file an issue?",
+        ok,
+        cancel
+    );
+
+    if (!result || result === cancel) {
+        return;
+    }
+
+    return await captureDiagnostics(ctx, false);
+}
+
+async function captureDiagnosticsMode(
+    ctx: WorkspaceContext,
+    allowMinimalCapture: boolean
+): Promise<"Minimal" | "Full" | undefined> {
+    if (
+        ctx.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0)) ||
+        vscode.workspace.getConfiguration("sourcekit-lsp").get<string>("trace.server", "off") !==
+            "off"
+    ) {
+        const fullButton = allowMinimalCapture ? "Capture Full Diagnostics" : "Capture Diagnostics";
+        const minimalButton = "Capture Minimal Diagnostics";
+        const buttons = allowMinimalCapture ? [fullButton, minimalButton] : [fullButton];
+        const fullCaptureResult = await vscode.window.showInformationMessage(
+            `A Diagnostic Bundle collects information that helps the developers of the VS Code Swift extension diagnose and fix issues.
```

Comment:
I think according to https://code.visualstudio.com/brand we should refer to it as *Swift for VS Code*

```suggestion
            `A Diagnostic Bundle collects information that helps the developers of the Swift for VS Code extension diagnose and fix issues.
```